### DISCUSSION TITLE
Impostare un ritardo dell'attivazione da pulsante di un attuatore Shelly

### DISCUSSION THREAD

**Discussion by Macretti** [indomus/forum#1053] [HUB: Home Assistant]
ciao a tutti, sono nuovo di HA e del forum ed è la prima volta che scrivo. 
ho la necessità di ritardare di qualche secondo l'attuazione di une relè (sto usando uno shelly 1 v3) tramite pulsante fisico. 

ho provato a fare così: settando il 'button type' dello shelly su 'momentary', ho creato un'automazione che a seguito della pressione del pulsante attiva lo stesso dispositivo dopo un ritardo impostato, poi settando 'button type' su 'detached switch' funziona, in quanto alla pressione fisica del pulsante il relè non scatta, ma l'automazione aziona cmq lo shelly dopo il ritardo. eureka! tuttavia... 
non appena HA 'si accorge' che l'entità pulsante dello shelly non esiste più (detached switch) fornisce ovviamente un errore e l'automazione smette di funzionare e diventa indisponibile. 

Cortesi suggestions? 
grazie

**Comment by snakuzzo** [comment]
Non so se mi è chiaro il tutto...ma...
Per caso usi l'integrazione nativa ?
Con quella probabilmente quando vai a cambiare impostazione lato shelly ti ricrea in maniera diversa l'entità.
Hai provato ad integrare il device via MQTT ?

**Reply by Macretti** [reply]
Grazie per la dritta! In effetti l'integrazione (nativa) di shelly su HA reimposta il dispositivo a seguito di modifiche effettuate 'lato shelly' e grazie a te ho trovato cmq la soluzione che cercavo!
Non ho ancora grande esperienza di MQTT, ho provato ad usarlo tempo fa e anche se intuisco che sia un protocollo più efficiente e 'robusto', non so ancora per quale motivo ho osservato dei tempi di reazione dei miei dispostivi piuttosto lunghi, ma non ho indagato più di tanto, lo farò prossimamente. Al momento uso i vari shelly integrati in local (con coloT in unicast, puntando direttamente all'IP del RPI che ospita HA) e per ora le risposte sono istantanee, non perdono un colpo e nessun problema ancora rilevato.

**Reply by snakuzzo** [reply]
Ottimo! :)

**Comment by Macretti** [comment]
Risolto! (O almeno credo...). Posto la soluzione del problema nel caso possa interessare a qualcun'altro. 

Mi spiego meglio con un esempio:

ho un pulsante, uno shelly 1 e una lampadina. L'obiettivo è accendere la lampadina dopo tot secondi dalla pressione del pulsante.
Usando l'integrazione nativa di shelly di HA, basta settare 'Button type' su 'detached switch' ('lato shelly', via browser insomma) e tra le entità dello shelly su HA  apparirà una nuova proprietà 'input' (inizialmente disabilitata dall'integrazione, basta abilitarla manualmente) che con qualche limitazione (è solo on/off) sostituisce l'entità 'pulsante' nel trigger dell'automazione che si andrà a creare e che avrà come azioni un ritardo di esecuzione e l'attivazione dello stesso shelly.
